### DISCUSSION TITLE
Write doc comments as file comments

### DISCUSSION THREAD

**Discussion by ShayBox** [cbeck88/conf-rs#21] [Ideas]
Could you parse the doc comments and write them to the config files by appending each writeln?

**Comment by cbeck88** [comment]
yeah in theory this is possible

**Comment by ShayBox** [comment]
Serde does not want to support this, and I tried to add it to my own derive proc macro crate, but I don't know enough to make it work recursively

**Reply by cbeck88** [reply]
the way i've tried to draw the boundary in this crate is, i'm interacting with deserializers, and i have no dependency on any specific format or deserializer implementation

in this crate, to continue that pattern, what i'd want to do is like, make a function on `Conf` that produces a "serializer" for the help of that structure. then my proc macro has to make a local type that implements serializer, and for each field, either needs to produce the doc string, or if its a flattened substructure, calls the "get serializer" function on that recursively

not totally sure how it should work for subcommands, maybe it can create a substructure for each one.

as the user, you would call this, get a serializer, and pass that to `serde_json` or `serde_yaml` or whatever and write it out as you please.

there's some ambiguity about what is the right feature though, and like, how would people configure this if they want slightly more than just the doc string

another option you have is to use something like utoipa I guess? https://docs.rs/utoipa/latest/utoipa/derive.ToSchema.html
if you derive `ToSchema` i think it will read all your doc strings and produce a JsonSchema, which is possibly even more useful to write out to disk

**Reply by ShayBox** [reply]
I'm not sure configuration is needed other than a way to opt-in like `#[conf(docs)]` or a feature flag, the user can chose between `///` doc strings, and `//` or `/* ... */` comments which aren't written to the file

**Comment by cbeck88** [comment]
I added a way to do this and an example here: 7679c5ee3889b30c242323173de54508b4610d9c, lmk what you think
